flink-table/flink-sql-jdbc-driver/src/main/java/org/apache/flink/table/jdbc/FlinkConnection.java

```diff
@@ -55,41 +85,75 @@ public DatabaseMetaData getMetaData() throws SQLException {
 
     @Override
     public void setCatalog(String catalog) throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        try {
+            setSessionCatalog(catalog);
+        } catch (Exception e) {
+            throw new SQLException(String.format("Set catalog[%s] fail", catalog), e);
+        }
+    }
+
+    private void setSessionCatalog(String catalog) {
+        executor.configureSession(String.format("USE CATALOG %s;", catalog));
     }
 
     @Override
     public String getCatalog() throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        try (StatementResult result = executor.executeStatement("SHOW CURRENT CATALOG;")) {
+            if (result.hasNext()) {
+                return result.next().getString(0).toString();
+            } else {
+                throw new SQLException("No catalog");
+            }
+        }
     }
 
     @Override
     public void setClientInfo(String name, String value) throws SQLClientInfoException {
-        throw new SQLClientInfoException();
+        executor.configureSession(String.format("SET '%s'='%s';", name, value));
     }
 
     @Override
     public void setClientInfo(Properties properties) throws SQLClientInfoException {
-        throw new SQLClientInfoException();
+        for (Object key : properties.keySet()) {
+            setClientInfo(key.toString(), properties.getProperty(key.toString()));
+        }
     }
 
     @Override
     public String getClientInfo(String name) throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        Configuration configuration = (Configuration) executor.getSessionConfig();
```

Comment:
Maybe the assumption is not always true in the future. (I'm big fun for coding against interface instead of implementation). Maybe we should improve `ReadableConfig` interface for reading configs dynamically.

Reply:
I agree with you and I think executor.getSessionConfig() should config without other dependencies such as `Map<String, String>`. I will add `TODO` here and fix it in https://issues.apache.org/jira/browse/FLINK-31687 . What do you think?